browser/mapping.go

```diff
@@ -688,14 +686,22 @@ func mapBrowserContext(vu moduleVU, bc *common.BrowserContext) mapping { //nolin
 				var runInTaskQueue func(p *common.Page) (bool, error)
 				if parsedOpts.PredicateFn != nil {
 					runInTaskQueue = func(p *common.Page) (bool, error) {
+						vu.tqMu.Lock()
```

Comment:
If I understand correctly, we use this `vu.tqMu` to sync access for all task queues used for every page independently. So here for example we use it as the same task queue for a given page might be used concurrently, for example in a handler registered through `page.on` and a predicate in `waitForEvent`. Is this correct?

Reply:
One more thing to add is that once a taskqueue is retrieved from the taskqueues, we're not protecting the access to that since it look thread safe. So i would expect that `tq.Queue(func() error {...` can be called from multiple threads on the same `tq` and that's ok. So `tqMu` is protecting the taskqueues data structure itself.

Reply:
Oh, but i think i know what you're getting at now though, there's no need to use `tqMu` when working with `tq.Queue`. That'll be fixed in the next commit with the taskqueue higher level API refactor.

Reply:
Fixed in 1e7d7b75cfcd5dc3405faf5ad7e3354ccad561fc

Reply:
Yep, so that was my doubt basically, whether `taskQueue` itself was actually thread safe, as we could be queuing tasks concurrently between different executions of `waitForEvent` and `on`. But [it is actually thread](https://github.com/mstoykov/k6-taskqueue-lib/blob/111459812ad27077dfadf4cd01c131019d24b7ae/taskqueue/queuer.go#L62-L63) safe within its implementation, so I think this change is correct.